src/main/java/iterator/Extra.java

```diff
@@ -13,7 +14,7 @@ private Extra() {
      * @implNote try implement it with only one line of code !
      */
     public static int sumAges(RichIterator<Person> itr) {
-        throw new NotImplementedException();
+        return itr.isEmpty() ? 0 : itr.map(person -> person.age).reduce(Integer::sum);
```

Comment:
Very nice, I would use `reduceOptional` instead of manually checking for empty but it is just my preference.
```suggestion
        return itr.map(person -> person.age).reduceOptional(Integer::sum).orElse(0);
```